build/cmake/CMakeModules/AddZstdCompilationFlags.cmake

```diff
@@ -2,24 +2,24 @@ include(CheckCXXCompilerFlag)
 include(CheckCCompilerFlag)
 
 function(EnableCompilerFlag _flag _C _CXX)
-    message("Checking flag ${_flag}")
+    string(REGEX REPLACE "[^A-Za-z0-9+]+" "_" varname "${_flag}")
```

Comment:
This is the part that causes the fail, at least with `2.8.12`.  It's because CMake tries to compile the var name as a regex inside of `CHECK_CXX_COMPILER_FLAG` (no clue why...) so "`..._STD_C++11`" fails because it's not valid.  Easiest way to fix it I think would just be to take the `+` out of the match group here so it's converted to a `_`.